src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsClient/DiagnosticsClient.cs

```diff
@@ -217,11 +217,22 @@ internal ProcessInfo GetProcessInfo()
         /// </returns>
         public static IEnumerable<int> GetPublishedProcesses()
         {
-            return Directory.GetFiles(PidIpcEndpoint.IpcRootPath)
-                .Select(namedPipe => (new FileInfo(namedPipe)).Name)
-                .Where(input => Regex.IsMatch(input, PidIpcEndpoint.DiagnosticsPortPattern))
-                .Select(input => int.Parse(Regex.Match(input, PidIpcEndpoint.DiagnosticsPortPattern).Groups[1].Value, NumberStyles.Integer))
-                .Distinct();
+            static IEnumerable<int> GetAllPublishedProcesses()
+            {
+                foreach (var namedPipe in Directory.GetFiles(PidIpcEndpoint.IpcRootPath))
+                {
+                    var fileName = new FileInfo(namedPipe).Name;
```

Comment:
```suggestion
                foreach (var port in Directory.GetFiles(PidIpcEndpoint.IpcRootPath))
                {
                    var fileName = new FileInfo(port).Name;
```

nit: We use sockets on Linux/Unix and NamedPipes on Windows, so let's use somethign more generic.